src/Cassandra.IntegrationTests/Mapping/Tests/InsertTests.cs

```diff
@@ -426,6 +427,30 @@ public void Insert_Without_Nulls_Test()
             Assert.AreEqual("Substitute 3", storedSong.Title);
         }
 
+        [Test]
+        public void Insert_With_TTL()
+        {
+            var config = new MappingConfiguration()
+                .Define(new Map<Song>().PartitionKey(s => s.Id).TableName("song_insert"));
+            //Use linq to create the table
+            new Table<Song>(_session, config).CreateIfNotExists();
+            var mapper = new Mapper(_session, config);
+            var song = new Song
+            {
+                Id = Guid.NewGuid(),
+                Artist = "The Who",
+                Title = "Substitute",
+                ReleaseDate = DateTimeOffset.UtcNow
+            };
+            mapper.Insert(song, true, 5);
+            var notExpiredSong = mapper.First<Song>("WHERE id = ?", song.Id);
+            Assert.NotNull(notExpiredSong);
+            Assert.AreEqual(song.Id, notExpiredSong.Id);
+            Assert.AreEqual(song.Artist, notExpiredSong.Artist);
+            Assert.AreEqual(song.Title, notExpiredSong.Title);
+            Thread.Sleep(6000);
+            Assert.Throws<InvalidOperationException>(() => mapper.First<Song>("WHERE id = ?", song.Id));
```

Comment:
Using `FirstOrDefault()` and checking for null would be more clear. `InvalidOperationException` could be thrown for multiple reasons.